File: client/src/app/House/Models/room-details.ts
import { Room } from './room';
import { SensorHistory } from './sensor-history';
import { DeviceDetails } from './device-details';

export interface RoomDetails {
    id: string,
    name: string,
    type: string,
    sensorsHistory: SensorHistory[],
    devices: DeviceDetails[]
}

File: server/SmartHome.BusinessLogic/Rooms/Models/RoomDetails.cs
using SmartHome.BusinessLogic.Components.Models;
using System;
using System.Collections.Generic;

namespace SmartHome.BusinessLogic.Rooms.Models
{
    public class RoomDetails
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public IReadOnlyCollection<SensorHistory> SensorsHistory { get; set; }
        public IReadOnlyCollection<DeviceDetails> Devices { get; set; }
    }
}
